packages/next/install-swc.js

```diff
@@ -0,0 +1,256 @@
+  const cacheDirectory = (() => {
+    let result
+    const envDefined = process.env['NEXT_SWC_PATH']
+
+    if (envDefined) {
+      result = envDefined
+    } else {
+      let systemCacheDirectory
+      if (process.platform === 'linux') {
+        systemCacheDirectory =
+          process.env.XDG_CACHE_HOME || path.join(os.homedir(), '.cache')
+      } else if (process.platform === 'darwin') {
+        systemCacheDirectory = path.join(os.homedir(), 'Library', 'Caches')
+      } else if (process.platform === 'win32') {
+        systemCacheDirectory =
+          process.env.LOCALAPPDATA ||
+          path.join(os.homedir(), 'AppData', 'Local')
+      } else {
+        console.error(new Error('Unsupported platform: ' + process.platform))
+        process.exit(0)
+      }
+      result = path.join(systemCacheDirectory, 'next-swc')
+    }
+
+    if (!path.isAbsolute(result)) {
+      // It is important to resolve to the absolute path:
+      //   - for unzipping to work correctly;
+      //   - so that registry directory matches between installation and execution.
+      // INIT_CWD points to the root of `npm/yarn install` and is probably what
+      // the user meant when typing the relative path.
+      result = path.resolve(process.env['INIT_CWD'] || process.cwd(), result)
+    }
+    return result
+  })()
+
+  const extractFromTar = async () => {
+    await fs.promises.mkdir(outputDirectory).catch((err) => {
+      if (err.code !== 'EEXIST') throw err
+    })
+    const tarFilepath = path.join(cacheDirectory, tarFileName)
+    const readStream = fs.createReadStream(tarFilepath)
+    const writeStream = fs.createWriteStream(outputFilepath)
+    let foundEntry = false
+
+    await new Promise((resolve, reject) => {
+      readStream
+        .pipe(tar.t())
+        .on('entry', (entry) => {
+          if (entry.path.endsWith('.node')) {
+            foundEntry = true
+            entry
+              .pipe(writeStream)
+              .on('error', (err) => reject(err))
+              .on('finish', () => resolve())
+          }
+        })
+        .on('error', (err) => reject(err))
+        .on('finish', () => {
+          if (!foundEntry) {
+            reject(new Error(`Failed to find entry in ${tarFilepath}`))
+          }
+        })
+    })
+    await rmFile(versionFilepath)
+    await fs.promises.writeFile(versionFilepath, version)
+  }
+
+  // check cache first if it exists copy from there
+  if (await exists(path.join(cacheDirectory, tarFileName))) {
+    await extractFromTar()
+    return
+  }
+
+  // download fresh copy and populate cache
+  const fetch = require('next/dist/compiled/node-fetch')
+
+  const tempFile = path.join(
+    cacheDirectory,
+    `${tarFileName}.temp-${Date.now()}`
+  )
+  await fs.promises.mkdir(cacheDirectory, { recursive: true })
+
+  try {
+    await fetch(
+      `https://registry.npmjs.org/${activePackage.packageName}/-/${tarFileName}`
```

Comment:
Some companies block direct access to the registry so this will be problematic
Ref https://github.com/evanw/esbuild/pull/1621

Reply:
Yeah this is problematic because package config like `.npmrc` or `.yarnrc` is no longer respected.

https://github.com/evanw/esbuild/issues/286

Reply:
We can definitely handle loading the registry from the package config. I think it's gonna be a while before this is sorted out/performant in relevant package managers so the custom install script will still be needed for a bit.